### PR TITLE
Core/dp 20309 overview documentation

### PR DESCRIPTION
**Pull request by tkoleary**
## Description
- Refactored the mdx Introduction page to not use the rendered component because the links would not work (inside iFrame). 
- Re-wrote the content to be terser and more up-to-date with the scope and purpose of Mayflower core.
- Added some styles to override sb classes leaking into the components

## Related Issue / Ticket
- [JIRA issue](DP-20309)

## Steps to Test
1. Switch to the branch
2. Run rush start:core
3. Go to http://localhost:6006/?path=/docs/overview-introduction--page
4. Compare to screenshot, including left nav

## Screenshots
Screen Shot 2020-12-07 at 11 48 48 AM

#### @TODO
Remove the `.sandbox` class and use the preview/head to add overrides or resets

### CODE REVIEW

File: packages/core/stories/about/Introduction.stories.mdx [hunk 1]
```diff
@@ -5,60 +5,104 @@ import {
...
-      title={{
-        text: 'Design Principles'
-      }}
-      description="Follow these high-level guides when making decisions on design and development of any digital or print products, and when adopting or contributing to Mayflower."
-      index={0}
-    >
-      <DecorativeLink text="Brand Pillars" href="" />
-      <DecorativeLink text="Design Guidelines" href="" />
-    </SectionLinks>
+    <section className="ma__section-links">
```
Comment (clairesunstudio) [inline packages/core/stories/about/Introduction.stories.mdx]: why not use the mayflower react component for SectionLinks?

File: packages/core/stories/principles/Accessibility.stories.mdx [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { Meta } from '@storybook/addon-docs/blocks';
 import generateTitle from '../util/generateTitle';
 
-<Meta title="Design Principles/Guidelines/Accessibility" />
+<Meta title="Principles/Accessibility" />
```
Comment (clairesunstudio) [inline packages/core/stories/principles/Accessibility.stories.mdx]:
```suggestion
<Meta title={generateTitle('Accessibility')} />
```

File: packages/core/stories/about/Introduction.stories.mdx (hunk 1 above)
Comment (tkoleary) [reply packages/core/stories/about/Introduction.stories.mdx]: When I did that I could not get links to other pages in docs to work. Is there a special syntax for that? Couldn't find any info in Storybook docs.

Comment (tkoleary) [reply packages/core/stories/about/Introduction.stories.mdx]: Ah. I found it. The documentation is in the addon

Comment (tkoleary) [reply packages/core/stories/about/Introduction.stories.mdx]: Ok, pushed a new commit with LinkTo implemented in the MDX file.

Comment (clairesunstudio) [reply packages/core/stories/about/Introduction.stories.mdx]: I think you could just use LinkTo inside of <SectionLinks>. Trying that now, if that works I will push it up

**Review by clairesunstudio — APPROVED** [review]

**Pull request merged**